app/notification.go

```diff
@@ -657,6 +753,44 @@ func (a *App) getMentionKeywordsInChannel(profiles map[string]*model.User, allow
 	return keywords
 }
 
+// insertGroupMentions adds Mentions for users that are part of the given group members and channel channel, adds group members that are not in the channel to OtherPotentialMentions
+// returns true if atleast one user from the group exists in the team that the channel belongs to.
```

Comment:
"exists in the team" - is it the team or the channel? I see that profileMap is built off GetAllProfilesInChannel() result.

Reply:
yea its exists in team cause it looks at users in the channel from `profileMap` and then it gets the list of `outOfChannelGroupMembers` so in the case where there are no `outOfChannelGroupMembers` and no users mentioned from the `profileMap` it returns `false` to let us know that the group mention did nothing

Reply:
I was referring to "return true" in the comment , not "return false". Unless I miss something, `userMentioned` is set to true if there is at least one user from the channel, irrespective of the other conditions.

Reply:
```
for _, user := range users {
		if _, ok := profileMap[user.Id]; ok {
			mentions.Mentions[user.Id] = GroupMention
			userMentioned = true
		}
	}
```

`users` refers to group members here - i think the variable should be renamed to `groupMembers`

So it iterates through all the group member users and then for each one checks if the `profileMap` matches a user id if theres atleast one user found it sets `userMentioned` to true